src/main/java/com/purple/ua/universityappointment/security/MyUserDetailsService.java

```diff
@@ -7,19 +7,23 @@
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.Optional;
 
-@Service
+@Service("MyUserDetailsService")
 public class MyUserDetailsService implements UserDetailsService {
 
     @Autowired
     UserRepository userRepository;
 
     @Override
+    @Transactional
     public UserDetails loadUserByUsername(String userLogin) throws UsernameNotFoundException {
         Optional<User> user = userRepository.findByLogin(userLogin);
         user.orElseThrow(() -> new UsernameNotFoundException("Not found:  " + userLogin));
         return user.map(MyUserDetails::new).get();
```

Comment:
if you had User, as i sad, there could be just one line: return userRepository.findByLogin(userLogin).orElseThrow(() -> new UsernameNotFoundException("Not found:  " + userLogin));